src/signer/noscrypt_cipher.hpp

```diff
@@ -0,0 +1,160 @@
+
+#include <memory>
+
+#include <noscrypt.h>
+#include <noscryptutil.h>
+
+namespace nostr
+{
+    namespace signer
+    {
+        class NoscryptCipherContext
+        {
+        private:
+            NCUtilCipherContext* _cipher;
```

Comment:
Best to use smart pointers in C++, when possible.

```suggestion
            std::shared_ptr<NCUtilCipherContext> _cipher;
```